ouroboros-network/src/Ouroboros/Network/BlockFetch.hs

```diff
@@ -152,15 +151,15 @@ data BlockFetchConsensusInterface peer header block m =
        -- with operational key certificates there are also cases where
        -- we would consider a chain of equal length to the current chain.
        --
-       plausibleCandidateChain :: ChainFragment block
-                               -> ChainFragment header -> Bool,
+       plausibleCandidateChain :: AnchoredFragment block
+                               -> AnchoredFragment header -> Bool,
 
        -- | Compare two candidate chains and return a preference ordering.
        -- This is used as part of selecting which chains to prioritise for
        -- downloading block bodies.
        --
-       compareCandidateChains  :: ChainFragment header
-                               -> ChainFragment header
+       compareCandidateChains  :: AnchoredFragment header
+                               -> AnchoredFragment header
```

Comment:
It's here and in `plausibleCandidateChain` where the assumption (or otherwise) about a common anchor point really comes up.